ntp-udp/src/interface_name.rs

```diff
@@ -48,10 +48,10 @@ impl InterfaceAddress {
     ///
     /// assumes a valid `libc::ifaddrs`
     unsafe fn from_libc_ifaddrs(info: &libc::ifaddrs) -> InterfaceAddress {
```

Comment:
At this point, should this function no longer be declared `unsafe`? It seems that the required invariant that `&libc::ifaddrs` contains a vali `libc::ifaddrs` is quite natural and something that should be enforced at creation time.

Reply:
Unfortunately, due to libc being a direct binding to the c library, it is possible to create invalid libc::ifaddrs values in safe rust code, hence the unsafe here.

Reply:
Out of curiosity, can you give me an example of safe Rust code that will generate an invalid `libc::ifaddrs` value?

Reply:
```
let mut buf: [i8;2] = [1,2];
let a = libc::ifaddrs {
    ifa_next: std::ptr::null_mut(),
    ifa_name: buf.as_mut_ptr(),
    ifa_flags: 0,
    ifa_addr: std::ptr::null_mut(),
    ifa_netmask: std::ptr::null_mut(),
    ifa_ifu: std::ptr::null_mut(),
    ifa_data: std::ptr::null_mut(),
};
```
The resulting a is not a valid ifaddr (ifa_name is not a valid C string), and will in fact trigger UB when passed to `from_libc_ifaddrs`.

Reply:
Fair enough!